### PR TITLE
init

### PR DESCRIPTION
**Pull request by MaksymKos**
https://maksymkos.github.io/react_carousel/

### CODE REVIEW

**Review by anastasiiavorobiova — CHANGES_REQUESTED** [review]
Great work! 
Screenshot 2023-08-30 at 12 52 34
Item width doesn't work correctly. Consider fixing it

**Review by IvanFesenko — CHANGES_REQUESTED** [review]
Check this case and fix it
https://github.com/mate-academy/react_carousel/assets/61904995/f697d3c5-4973-43ab-bf2e-c6ff9cf3454a

Infinite scroll do not works on frame size 7+
https://github.com/mate-academy/react_carousel/assets/61904995/13eed08f-0292-4109-8f4a-3d5ceb940262

File: src/App.scss [hunk 1]
```diff
@@ -1 +1,39 @@
-// style go here
+* {
+  margin: 0;
+  padding: 0;
+  box-sizing: border-box;
+
+  font-family: 'Kanit', sans-serif;
+
+
```
Comment (loralevitska) [inline src/App.scss:8]: Remove extra empty lines

File: src/components/Carousel.scss [hunk 2]
```diff
@@ -1 +1,61 @@
...
+  display: flex;
+  align-items: center;
+  justify-content: center;
+
+  background-image: linear-gradient(144deg,#AF40FF, #5B42F3 50%,#00DDEB);
+  border: 0;
+  border-radius: 8px;
+  box-shadow: rgba(151, 65, 252, 0.2) 0 15px 30px -5px;
+  box-sizing: border-box;
+  color: #FFFFFF;
```
Comment (loralevitska) [inline src/components/Carousel.scss:50]: It is a good practice to make all colors [variable](https://developer.mozilla.org/en-US/docs/Web/CSS/Using_CSS_custom_properties)

File: src/components/Carousel.tsx [hunk 3]
```diff
@@ -1,18 +1,179 @@
...
+        <ul
+          className="Carousel__list"
+        >
+          {images.map(image => (
+            <li
+              key={image.id}
+              style={{
+                transform: `translateX(${-params.itemWidth * (firstVisibleImg - 1)}px)`,
+                transition: `transform ${params.animationDuration}ms`,
+              }}
```
Comment (loralevitska) [inline src/components/Carousel.tsx:73]: Move to variable

File: src/components/Carousel.tsx [hunk 4]
```diff
@@ -1,18 +1,179 @@
...
+          type="button"
+          onClick={scrolLeft}
+        >
+          Prev
+        </button>
+        <button
+          className="button-63"
+          disabled={(!params.infinite
+            && (firstVisibleImg === (images.length - params.frameSize + 1)))
+            || params.frameSize === images.length}
```
Comment (loralevitska) [inline src/components/Carousel.tsx:101]: Long expression move to variable. Fix all cases

**Review by loralevitska — CHANGES_REQUESTED** [review]
Almost everything is perfect) when the "next" button is pressed, the 7th picture cannot be seen, it seems to jump and does not fall into the frame at step 3
https://github.com/mate-academy/react_carousel/assets/70009205/32afcdc9-05ad-434d-a912-c6fc73e5d800

**Review by VitaliyBodnar — CHANGES_REQUESTED** [review]
- All tests are not passed
Screenshot 2023-08-31 at 13 45 14

**Comment by MaksymKos** [comment]
u can fix tests and after write it)

**Review by IvanFesenko — CHANGES_REQUESTED** [review]
Why step for size is 130?  What is that?
https://github.com/mate-academy/react_carousel/assets/61904995/03205f67-97ef-4ac3-bd07-18b986dfe383

File: src/components/Carousel.scss [hunk 5]
```diff
@@ -1 +1,59 @@
...
+  display: flex;
+  align-items: center;
+  justify-content: center;
+  background-image: linear-gradient(144deg,#AF40FF, #5B42F3 50%,#00DDEB);
+  border: 0;
+  border-radius: 8px;
+  box-shadow: rgba(151, 65, 252, 0.2) 0 15px 30px -5px;
+  box-sizing: border-box;
+  color: $button-text-color;
+  font-family: 'Courier New', Courier, monospace;
```
Comment (TarasHoliuk) [inline src/components/Carousel.scss:50]: It's not critical issues, but on the other hand there are some lint rules, so we shouldn't ignore them
It's easy to fix - please, do it :)
image

**Review by TarasHoliuk — CHANGES_REQUESTED** [review]
GJ :-1: 
Fix Lint errors

File: src/App.tsx [hunk 6]
```diff
@@ -1,39 +1,51 @@
...
+  frameSize: 3,
+  itemWidth: 130,
+  animationDuration: 1000,
+  infinite: false,
+};
+
+function getPreparedImages(img: Image[], filterParams: Params): Image[] {
+  const firstImgIndex = img.findIndex(el => el.id === filterParams.firstImg);
+
+  return [...img].slice(firstImgIndex);
```
Comment (etojeDenys) [inline src/App.tsx:22]: slice return the new array, so copying the array is redundant here

**Review by etojeDenys — APPROVED** [review]
good job

**Pull request closed without merging**